src/Symfony/Component/Serializer/NameConverter/MetadataAwareNameConverter.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\NameConverter;
+
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+final class MetadataAwareNameConverter implements AdvancedNameConverterInterface
+{
+    /**
+     * @var ClassMetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    public function __construct(ClassMetadataFactoryInterface $metadataFactory)
+    {
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($propertyName, string $class = null, string $format = null, array $context = array())
+    {
+        if (null === $class || !$this->metadataFactory->hasMetadataFor($class)) {
+            return $propertyName;
+        }
+
+        $classMetadata = $this->metadataFactory->getMetadataFor($class);
+
+        $attributesMetadata = $classMetadata->getAttributesMetadata();
```

Comment:
`return $classMetadata->getAttributesMetadata()[$propertyName]->getSerializedName() ?? $propertyName;` should do the trick